src/uri/authority.rs

```diff
@@ -53,6 +53,37 @@ impl Authority {
         })
     }
 
+    /// Attempt to convert an `Authority` from a static string.
+    ///
+    /// This function will not perform any copying, and the string will be
+    /// checked if it is empty or contains an invalid character.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the argument contains invalid characters or
+    /// is empty.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use http::uri::Authority;
+    /// let authority = Authority::from_static("example.com");
+    /// assert_eq!(authority.host(), "example.com");
+    /// ```
+    pub fn from_static(src: &'static str) -> Self {
+        let s = src.as_bytes();
+        let b = Bytes::from_static(s);
+        let authority_end = Authority::parse_non_empty(&b[..]).unwrap();
+
+        if authority_end != b.len() {
+            panic!("invalid uri character");
```

Comment:
This happens for things like `Authority::from_static("hyper.rs/guides")`, right? So, what should the message say? Same as the `expect` above?